File: bug24_wrong_parameter_order/solved_main.js
function weighted_mix(base, weight, offset) {
    let acc=base;
    for(let i=0n;i<10n;i++){
        acc^=(weight*i+offset);
        acc=BigInt.asUintN(64,acc*0x9E3779B97F4A7C15n);
        acc=BigInt.asUintN(64,(acc<<(i%7n+1n))|(acc>>(63n-i%7n)));
    }
    return acc;
}

function run() {
    let base_val=0xCAFEBABE12345678n, weight_val=0x100n, offset_val=0xDEADn;
    // FIX: was weighted_mix(weight_val, base_val, offset_val) — args were swapped
    let result = weighted_mix(base_val, weight_val, offset_val);
    let buf=Buffer.alloc(8);
    buf.writeBigUInt64BE(result,0);
    console.log(buf.toString('base64'));
}
run();

File: bug24_wrong_parameter_order/solved_main.java
import java.nio.ByteBuffer;
import java.util.Base64;

public class solved_main {
    static long weighted_mix(long base, long weight, long offset) {
        long acc=base;
        for(long i=0;i<10;i++){
            acc=(acc^(weight*i+offset))*0x9E3779B97F4A7C15L;
            acc=(acc<<(i%7+1))|(acc>>>(63-i%7));
        }
        return acc;
    }

    public static void main(String[] args) {
        long base_val=0xCAFEBABE12345678L, weight_val=0x100L, offset_val=0xDEADL;
        // FIX: was weighted_mix(weight_val, base_val, offset_val) — args were swapped
        long result = weighted_mix(base_val, weight_val, offset_val);
        ByteBuffer buf=ByteBuffer.allocate(8);
        buf.putLong(result);
        System.out.println(Base64.getEncoder().encodeToString(buf.array()));
    }
}
